src/tools/server.ml

```diff
@@ -209,6 +209,7 @@ let make_server_thread
         Lwt.catch
           (fun () ->
            Lwt_unix.accept listening_socket >>= fun (plain_fd, cl_socket_address) ->
+           Lwt_unix.setsockopt listening_socket Unix.SO_KEEPALIVE true;
```

Comment:
Pushed a few more commits to make the keepalive behaviour configurable.
Also fixed 2 file descriptor leaks (in error paths).
Can you review?